thenewboston_node/business_logic/storages/file_system.py

```diff
@@ -173,32 +187,32 @@ def _compress(self, absolute_file_path: Path) -> Path:
         with open(absolute_file_path, 'rb') as fo:
             original_data = fo.read()
 
-        logger.debug('File %s size: %s bytes', absolute_file_path, len(original_data))
-        best_absolute_file_path = absolute_file_path
-        best_data = original_data
-
-        for compressor in self.compressors:
-            compress_function = COMPRESSION_FUNCTIONS[compressor]
-            compressed_data = compress_function(original_data)  # type: ignore
-            compressed_size = len(compressed_data)
-            logger.debug(
-                'File %s compressed with %s size: %s bytes (%.2f ratio)', absolute_file_path, compressor,
-                compressed_size, compressed_size / len(original_data)
-            )
-            # TODO(dmu) LOW: For compressed_size == best[0] choose fastest compression
-            if compressed_size < len(best_data):
-                best_absolute_file_path = Path(str(absolute_file_path) + '.' + compressor)
-                best_data = compressed_data
-                logger.debug('New best %s: %s size', best_absolute_file_path, len(best_data))
-
-        if best_absolute_file_path != absolute_file_path:
-            logger.debug('Writing compressed file: %s (%s bytes)', best_absolute_file_path, len(best_data))
-            self._write_file(best_absolute_file_path, best_data, mode='wb')
+        original_size = len(original_data)
+        logger.debug('File %s size: %s bytes', absolute_file_path, original_size)
+        best_compressed_data = min(self._yield_compressed_data(original_data, compressors), key=lambda x: x.size)
+        if self.always_compress or best_compressed_data.size < original_size:
+            compressed_file_path = Path(str(absolute_file_path) + '.' + best_compressed_data.compressor)
+            logger.debug('Writing compressed file: %s (%s bytes)', compressed_file_path, len(best_compressed_data))
+            self._write_file(compressed_file_path, best_compressed_data.data, mode='wb')
 
             logger.debug('Removing %s', absolute_file_path)
             os.remove(absolute_file_path)
+            return compressed_file_path
+
+        return absolute_file_path
 
-        return best_absolute_file_path
+    @staticmethod
+    def _yield_compressed_data(original_data, compressors) -> Generator[CompressedData, None, None]:
```

Comment:
Changed the name to `compressed data` from `compressed meta`

Reply:
does not seem that you did